aten/src/ATen/cpu/vec/vec256/vec256_convert.h

```diff
@@ -314,6 +314,23 @@ struct VecConvert<float, 1, BFloat16, 1> {
     return result;
   }
 };
+
+template <>
+struct VecConvert<BFloat16, 1, float, 2> {
+  static inline VectorizedN<BFloat16, 1> apply(
+      const VectorizedN<float, 2>& src) {
+    VectorizedN<BFloat16, 1> result;
+    uint32x4x2_t u32x4x2_0 = vld1q_u32_x2(reinterpret_cast<const uint32_t*>(&src[0]));
+    uint32x4x2_t u32x4x2_1 = vld1q_u32_x2(reinterpret_cast<const uint32_t*>(&src[1]));
+    uint16x4_t u16x4_0 = vmovn_u32(vshrq_n_u32(u32x4x2_0.val[0], 16));
```

Comment:
```suggestion
    uint16x4_t u16x4_0 =vshrn_n_u32(u32x4x2_0.val[0], 16);
```

You can use narrowing shift to avoid vmovn. ARM is extremely rich in types of instructions it supports but it is also confusing because there can always be more compact way of doing that.